iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp

```diff
@@ -54,245 +54,272 @@ class ChunkQueue_testBase
+    this->m_pusher.push(chunk);
 
     EXPECT_THAT(condVarWaiter.timedWait(1_ns), Eq(true));
     EXPECT_THAT(condVarWaiter.timedWait(1_ns), Eq(false)); // shouldn't trigger a second time
 }
 
-TEST_P(ChunkQueue_test, DISABLED_AttachSecondConditionVariableSignaler)
+TYPED_TEST(ChunkQueue_test, AttachSecondConditionVariableSignaler)
 {
     ConditionVariableData condVar1;
     ConditionVariableData condVar2;
     ConditionVariableWaiter condVarWaiter1{&condVar1};
     ConditionVariableWaiter condVarWaiter2{&condVar1};
 
-    auto ret1 = m_popper.attachConditionVariable(&condVar1);
+    auto ret1 = this->m_popper.attachConditionVariable(&condVar1);
     EXPECT_TRUE(ret1);
 
-    auto ret2 = m_popper.attachConditionVariable(&condVar2);
+    auto ret2 = this->m_popper.attachConditionVariable(&condVar2);
     EXPECT_FALSE(ret2);
 
     EXPECT_THAT(condVarWaiter1.timedWait(1_ns), Eq(false));
     EXPECT_THAT(condVarWaiter2.timedWait(1_ns), Eq(false));
 
-    auto chunk = allocateChunk();
-    m_pusher.push(chunk);
+    auto chunk = this->allocateChunk();
+    this->m_pusher.push(chunk);
 
     EXPECT_THAT(condVarWaiter1.timedWait(1_ms), Eq(true));
     EXPECT_THAT(condVarWaiter2.timedWait(1_ms), Eq(false));
 }
 
 /// @note this could be changed to a parameterized ChunkQueueSaturatingFIFO_test when there are more FIFOs available
+using ChunkQueueFiFoTestSubjects = Types<ThreadSafePolicy, SingleThreadedPolicy>;
+/// we require TYPED_TEST since we support gtest 1.8 for our safety targets
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+TYPED_TEST_CASE(ChunkQueueFiFo_test, ChunkQueueFiFoTestSubjects);
+#pragma GCC diagnostic pop
+
+template <typename PolicyType>
 class ChunkQueueFiFo_test : public Test, public ChunkQueue_testBase
 {
   public:
     void SetUp() override{};
     void TearDown() override{};
 
-    using ChunkQueueData_t = ChunkQueueData<iox::DefaultChunkQueueConfig>;
+    using ChunkQueueData_t = ChunkQueueData<iox::DefaultChunkQueueConfig, PolicyType>;
 
     ChunkQueueData_t m_chunkData{iox::cxx::VariantQueueTypes::FiFo_SingleProducerSingleConsumer};
     ChunkQueuePopper<ChunkQueueData_t> m_popper{&m_chunkData};
     ChunkQueuePusher<ChunkQueueData_t> m_pusher{&m_chunkData};
 };
 
 /// @note API currently not supported
-TEST_F(ChunkQueueFiFo_test, DISABLED_InitialSize)
+TYPED_TEST(ChunkQueueFiFo_test, DISABLED_InitialSize)
 {
-    EXPECT_THAT(m_popper.size(), Eq(0u));
+    EXPECT_THAT(this->m_popper.size(), Eq(0u));
 }
 
 /// @note API currently not supported
-TEST_F(ChunkQueueFiFo_test, DISABLED_Capacity)
+TYPED_TEST(ChunkQueueFiFo_test, DISABLED_Capacity)
 {
-    EXPECT_THAT(m_popper.getCurrentCapacity(), Eq(iox::MAX_RECEIVER_QUEUE_CAPACITY));
+    EXPECT_THAT(this->m_popper.getCurrentCapacity(), Eq(iox::MAX_RECEIVER_QUEUE_CAPACITY));
 }
 
-
 /// @note API currently not supported
-TEST_F(ChunkQueueFiFo_test, DISABLED_SetCapacity)
+TYPED_TEST(ChunkQueueFiFo_test, DISABLED_SetCapacity)
```

Comment:
Why is it disabled?

Reply:
See your comment above the test `/// @note API currently not supported`